### DISCUSSION TITLE
unable to locate the GNB application

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#95] [General Help]
raju@raju-ThinkPad-P16s-Gen-1:~/srsran/srsRAN_Project/build/apps/gnb$ ls -lrt
total 20
-rw-rw-r-- 1 raju raju 6401 Jun  9 14:59 Makefile
-rw-rw-r-- 1 raju raju  301 Jun  9 14:59 CTestTestfile.cmake
-rw-rw-r-- 1 raju raju 1138 Jun  9 14:59 cmake_install.cmake
drwxrwxr-x 2 raju raju 4096 Jun  9 14:59 CMakeFiles

normally i used to have gnb application here itself after i have formatted my Ubuntu and then i have again installed the srsran then i was unable to locate the gnb application

**Comment by brendan-mcauliffe** [comment]
Hi @rajugbv, 

If the application is not present after you rebuilt the code, then there was most likely an issue and it did not build correctly. I suggest you try to build the code again paying close attention that there are no errors. Also, double-check that you have reinstalled the dependencies correctly after formatting your PC.

**Comment by rajugbv** [comment]
@brendan-mcauliffe thank you i think i was getting error while running the cmale ../  command 

raju@raju-ThinkPad-P16s-Gen-1:~/srsRAN_Project/build$ cmake ../
-- Build type not specified: defaulting to Release.
-- Assertion level set to NORMAL
-- Could NOT find libdw (missing: LIBDW_LIBRARY LIBDW_INCLUDE_DIR) 
-- Could NOT find libbfd (missing: LIBBFD_LIBRARY LIBBFD_INCLUDE_DIR) 
-- Could NOT find libdwarf (missing: LIBDWARF_LIBRARY LIBDWARF_INCLUDE_DIR LIBELF_LIBRARY LIBELF_INCLUDE_DIR) 
-- Backward-cpp found, but external libraries are missing.
-- Checking for module 'mbedtls'
--   No package 'mbedtls' found
-- MBEDTLS LIBRARIES: /usr/lib/x86_64-linux-gnu/libmbedcrypto.so
-- MBEDTLS STATIC LIBRARIES: /usr/lib/x86_64-linux-gnu/libmbedcrypto.a
-- MBEDTLS INCLUDE DIRS: /usr/include
-- FFTW3F LIBRARIES: /usr/lib/x86_64-linux-gnu/libfftw3f.so
-- FFTW3F STATIC LIBRARIES: /usr/lib/x86_64-linux-gnu/libfftw3f.a
-- FFTW3F INCLUDE DIRS: /usr/include
-- UHD LIBRARIES /usr/local/lib/libuhd.so
-- UHD INCLUDE DIRS /usr/local/include
-- YAMLCPP LIBRARIES: /usr/lib/x86_64-linux-gnu/libyaml-cpp.so
-- YAMLCPP INCLUDE DIRS: /usr/include
-- SSE4.1 is enabled - target CPU must support it
-- AVX is enabled - target CPU must support it
-- AVX2 is enabled - target CPU must support it
-- FMA is enabled - target CPU must support it
-- Checking for module 'sctp'
--   No package 'sctp' found
-- SCTP LIBRARIES: /usr/lib/x86_64-linux-gnu/libsctp.so
-- SCTP INCLUDE DIRS: /usr/include
-- FFT_LIBRARIES: /usr/lib/x86_64-linux-gnu/libfftw3f.so
-- Building srsRAN version 23.5.0
-- Configuring done
-- Generating done
-- Build files have been written to: /home/raju/srsRAN_Project/build

how can i resolve this error ?
-- Checking for module 'sctp'
--   No package 'sctp' found

**Comment by herlesupreeth** [comment]
But it finds SCTP library, see log

> -- SCTP LIBRARIES: /usr/lib/x86_64-linux-gnu/libsctp.so
> -- SCTP INCLUDE DIRS: /usr/include

**Reply by rajugbv** [reply]
oh okay , thank you for response

**Comment by herlesupreeth** [comment]
Are you sure you installed all the dependencies in this link ?? - https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html#build-tools-and-dependencies

**Reply by rajugbv** [reply]
yes i did,  as i am was using Ubuntu i have run this command "sudo apt-get install cmake make gcc g++ pkg-config libfftw3-dev libmbedtls-dev libsctp-dev libyaml-cpp-dev libgtest-dev"

**Comment by robertfalkenberg** [comment]
Note that  `cmake ../` only prepares a bunch of Makefiles in the build directory. To do the actual build, you have to run `make` or `make -j NUM` with NUM being the number of CPUs.

**Comment by rajugbv** [comment]
**i have installed open5gs then changed config setting of amf.yml upf.yml**

raju@raju-ThinkPad-P16s-Gen-1:~/open5gs/configs/open5gs$ diff upf.yaml.in upf.yaml.in_org
201c201
<       - addr: 10.53.1.2  
---
>       - addr: 127.0.0.6 
raju@raju-ThinkPad-P16s-Gen-1:~/open5gs/configs/open5gs$ diff amf.yaml.in amf.yaml.in_org
474c474
<       - addr: 127.0.0.5 
---
>       - addr: 127.0.0.5
477c477
<       - addr: 10.53.1.2 
---
>       - addr: 127.0.0.5
483,484c483,484
<           mcc: 001
<           mnc: 01 
---
>           mcc: 999
>           mnc: 70
490,492c490,492
<           mcc: 001 
<           mnc: 01 
<         tac: 7
---
>           mcc: 999
>           mnc: 70
>         tac: 1
495,496c495,496
<           mcc: 001 
<           mnc: 01 
---
>           mcc: 999
>           mnc: 70
 **this are the file changes i have done in /home/raju/open5gs/configs/open5gs path

in srsran this are the changes i have done** 

amf:
  addr:   10.53.1.2                                              # The address or hostname of the AMF.
  bind_addr: 127.0.0.1                                            # A local IP that the gNB binds to for traffic from the AMF.

ru_sdr:
  device_driver: uhd                                            # The RF driver name.
  device_args: type=b200,num_recv_frames=64,num_send_frames=64  # Optionally pass arguments to the selected RF driver.
  clock: internal                                                        # Specify the clock source used by the RF.
  sync:  internal                                                       # Specify the sync source used by the RF.
  srate: 23.04                                                  # RF sample rate might need to be adjusted according to selected bandwidth.
  otw_format: sc12
  tx_gain: 50                                                   # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 60                                                   # Receive gain of the RF might need to adjusted to the given situation.

cell_cfg:
  dl_arfcn: 632628                                                # ARFCN of the downlink carrier (center frequency).
  band: 78                                                        # The NR band.
  channel_bandwidth_MHz: 20                                       # Bandwith in MHz. Number of PRBs will be automatically derived.
  common_scs: 30                                                  # Subcarrier spacing in kHz used for data.
  plmn: "00101"                                                   # PLMN broadcasted by the gNB.
  tac: 7                                                          # Tracking area code (needs to match the core configuration).
  pci: 1                                                          # Physical cell ID.

log:
  filename: /tmp/gnb.log                                          # Path of the log file.
  all_level: warning                                              # Logging level applied to all layers.

pcap:
  mac_enable: false                                               # Set to true to enable MAC-layer PCAPs.
  mac_filename: /tmp/gnb_mac.pcap                                 # Path where the MAC PCAP is stored.
  ngap_enable: false                                              # Set to true to enable NGAP PCAPs.
  ngap_filename: /tmp/gnb_ngap.pcap                               # Path where the NGAP PCAP is stored.
~                                                                                                       

**so when i tried to run the gnb application i am getting this error below
. is there any problem with config or  there is issue with srsran application?**

raju@raju-ThinkPad-P16s-Gen-1:~/srsRAN_Project/build/apps/gnb$ sudo ./gnb -c /home/raju/srsRAN_Project/configs/gnb_rf_b200_tdd_n78_20mhz.yml
Lower PHY in dual executor mode.

--== srsRAN gNB (commit 56a771dfe) ==--

Connecting to AMF on 10.53.1.2:38412
Failed to connect SCTP socket to 10.53.1.2:38412. error="Connection timed out" timeout=1385ms
srsGNB ERROR: Failed to create SCTP gateway.
Aborted

**Comment by herlesupreeth** [comment]
Start your AMF first then start gNB so that a SCTP connection can be established

**Reply by rajugbv** [reply]
@herlesupreeth  thank you for responding ,so when i tried to start amf service in open5gs i was getting error like this do you any idea how to resolve this

raju@raju-ThinkPad-P16s-Gen-1:~/open5gs/configs/open5gs$ sudo open5gs-amfd
[sudo] password for raju: 
Open5GS daemon v2.6.4

06/22 11:33:12.846: [app] INFO: Configuration: '/etc/open5gs/amf.yaml' (../lib/app/ogs-init.c:126)
06/22 11:33:12.846: [app] INFO: File Logging: '/var/log/open5gs/amf.log' (../lib/app/ogs-init.c:129)
Failed to bind to port 0: Address already in use
06/22 11:33:12.848: [metrics] ERROR: Cannot start Prometheus HTTP server (../lib/metrics/prometheus/context.c:282)
06/22 11:33:12.848: [sbi] INFO: NF Service [namf-comm] (../lib/sbi/context.c:1438)
06/22 11:33:12.848: [sock] ERROR: socket bind(2) [127.0.0.5]:7777 failed (98:Address already in use) (../lib/core/ogs-socket.c:114)
06/22 11:33:12.848: [sock] ERROR: tcp_server() [127.0.0.5]:7777 failed (98:Address already in use) (../lib/core/ogs-tcp.c:72)
06/22 11:33:12.848: [sbi] ERROR: Cannot start SBI server (../lib/sbi/nghttp2-server.c:371)
06/22 11:33:12.848: [sctp] ERROR: Failed to intialize AMF (../src/amf/app.c:30)
06/22 11:33:12.848: [app] FATAL: Open5GS initialization failed. Aborted (../src/main.c:219)

**Reply by herlesupreeth** [reply]
It looks like an issue in your AMF configuration file. Please go through this guide extensively - https://open5gs.org/open5gs/docs/guide/01-quickstart/ to configure your AMF correctly

**Reply by rajugbv** [reply]
thank you i was able to establish gnb
